common/src/stack/pylib/stack/gen.py

```diff
@@ -872,3 +872,129 @@ def generate_bash(self):
 		profile  = [ '#! /bin/bash' ]
 		profile.extend(self.shellSection.generate())
 		return profile
+
+	def generate_ansible(self):
+		playbook = []
+		playbook.extend(self.shellSection.generate())
+		return playbook
+
+
+class AnsibleProfileTraversor(MainTraversor):
+	def pre(self):
+		self.tasks = {
+			'install-post': [],
+			'boot-pre': [],
+			'boot-post': []
+		}
+
+	def traverse_stack_script(self, node):
+		stage  = self.getAttr(node, 'stack:stage',  default='install-post')
+		chroot = self.getAttr(node, 'stack:chroot', default='true')
+
+		# QUESTION: Do we want to ignore these stages in Ansible?
+		if stage in [ 'install-pre', 'install-pre-package' ] or not chroot == 'true':
+			return False
+
+		def tabs_to_spaces(match):
+			"""Replace each match char (a tab) with 4 spaces"""
+			return match.group(1) + ' ' * 4 * len(match.group(2))
+
+		node_id = self.getAttr(node, 'stack:id')
+		shell  = self.getAttr(node, 'stack:shell', default='/bin/bash')
+		node_file = os.path.basename(self.getAttr(node, 'stack:file'))
+		script = self.collect(node).strip()
+
+		# Make a nice name for our Ansible task
+		name = f"{stage}: Create script {node_id}"
+		if node_file:
+			name += f" from {node_file}"
+
+		# Format the script to be valid yaml by replacing tabs with spaces
+		formatted_script = []
+		last_line_blank = True
+		for line in script.splitlines():
+			# Collapse down multiple blank lines
+			if not line and last_line_blank:
+				continue
+
+			# Replace beginning tabs with spaces
+			line = re.sub(r'^( *)(\t+)', tabs_to_spaces, line)
+
+			# Strip any trailing whitespace
+			line = line.rstrip()
+
+			# Indent it for the Ansible output
+			formatted_script.append(' '*8 + line)
+
+			# Record if we were blank, so we can collapse
+			last_line_blank = not bool(line)
```

Comment:
I don't think the `bool` is needed here.

Reply:
You're right. I kinda liked that the `bool` made it a little more obvious what I was testing for, but perhaps `last_line_blank = not line` is just as obvious. Thoughts @bsanders ?

Reply:
:x: bool